slide-deck.js

```diff
@@ -98,57 +112,68 @@ class slideDeck extends HTMLElement {
 
   // dynamic
   store = {};
+  slides;
+  slideNotes;
+  slideCanvas;
   slideCount;
+  activeSlide;
   controlPanel;
   eventButtons;
   viewButtons;
-  activeSlide;
-  body;
+  #body;
 
   // callbacks
   attributeChangedCallback(name, oldValue, newValue) {
     this[slideDeck.attrToPropMap[name]] = newValue || this.hasAttribute(name);
 
     switch (name) {
       case 'key-control':
-        this.keyControlChange();
+        this.#keyControlChange();
         break;
       case 'follow-active':
-        this.followActiveChange();
+        this.#followActiveChange();
+        this.#updateEventButtons();
+        break;
+      case 'hide-parts':
+        this.#hidePartsChange();
+        this.#updatePartButtons();
         break;
       case 'slide-view':
-        this.updateViewButtons();
+        this.#updateViewButtons();
         this.scrollToActive();
         break;
       default:
         break;
     }
 
-    this.updateEventButtons();
+    this.#updateEventButtons();
   }
 
   constructor() {
     super();
 
     // shadow dom and ID
     slideDeck.appendShadowTemplate(this);
-    this.setDeckID();
+    this.#setDeckID();
 
     // relevant nodes
-    this.body = document.querySelector('body');
+    this.#body = document.querySelector('body');
     this.controlPanel = this.querySelector(`[slot="control-panel"]`) ??
       this.shadowRoot.querySelector(`[part="control-panel"]`);
+    this.slides = this.querySelectorAll(':scope > :not([slot])');
+    this.slideNotes = this.querySelectorAll('[slide-note]');
+    this.slideCanvas = this.querySelectorAll('[slide-canvas]:not(:scope > *)');
```

Comment:
Do we want to assert that the notes and canvas are inside a slide? That there is only a single instance of each inside each slide?